### DISCUSSION TITLE
About ImageRegistrator interference

### DISCUSSION THREAD

**Discussion by artths** [kornia/kornia#2489] [Questions]
*Inference 
As far as I know ImageRegistrator module is a little network that is trained using PyTorch every time when you use it. 
Is there a way how I can use this in my network during inference time? Is it really neccesary to carry PyTorch and other stuff with it? I want to run it on the edge device.

**Comment by edgarriba** [comment]
What's your target platform ? Pytorch is a requirement here since we solve an online optimization process to get the transformation matrix between two given image views using automatic differentiation (Pytorch autograd). It's not a standard network that "generalizes, instead you need to "learn" every time the transformation. Thoughts @ducha-aiki ?

**Reply by artths** [reply]
I want it to work on Jetson Orin Nano. Although you can install PyTorch on it, the training process will be much slower than a trained and optimized network using TensorRT. I'm aiming for at least 10 FPS and running PyTorch I'm unlikely to achieve that.

**Reply by ducha-aiki** [reply]
That is basically Lukas-Kanade like procedure, so you don’t need any autograd in PyTorch. But autograd makes it easier to implement, otherwise you have to implement optimization by hand.
